src/components/cylc/Dot.vue

```diff
@@ -0,0 +1,88 @@
+<style>
+@import '~@/styles/status-indicator/resize.css';
+</style>
+
+<template>
+  <!--
+  The first group of classes below set the dot colour based on the task state.
+  The second group start, upon a state (hence colour) change for that task, and
+  stop, upon a user click onto that dot, the pulse animation.
+  -->
+  <status-indicator
+    :active="isActiveMappedState()"
+    :positive="isPositiveMappedState()"
+    :negative="isNegativeMappedState()"
+    :intermediary="isIntermediaryMappedState()"
+
+    v-on:click="dismissPulse()"
+    :pulse="isPulseOn"
+  >
+  </status-indicator>
+</template>
+
+<script>
+  // This script customises the default 'dot' icons and animations on them:
+  import 'status-indicator/styles.css'
+
+  export default {
+    name: 'Dot',
+    props: ['dotClass'],
+    watch: {
+      // Start the pulse animation if the task state (hence dot class) changes
+      hasDotClassChanged: function() {
+        this.isPulseOn = true
+      }
+    },
+    data: () => ({
+      // Status indicator icon class to replace each task state with:
+      stateToDotClassMappings: {
+        'runahead': 'intermediary',
+        'waiting': 'intermediary',
+        'held': 'intermediary',
+        'queued': 'intermediary',
+        'expired': 'negative',
+        'submitFailed': 'negative',
+        'failed': 'negative',
+        'ready': '',
+        'submitted': '',
+        'submitRetrying': 'active',
+        'retrying': 'active',
+        'running': 'active',
+        'succeeded': 'positive'
+      },
+      isPulseOn: false  // initialise at false
+    }),
+    methods: {
+      isActiveMappedState() {
+        return this.testInState(this.dotClass, 'active');
+      },
+      isPositiveMappedState() {
+        return this.testInState(this.dotClass, 'positive');
+      },
+      isNegativeMappedState() {
+        return this.testInState(this.dotClass, 'negative');
+      },
+      isIntermediaryMappedState() {
+        return this.testInState(this.dotClass, 'intermediary');
+      },
+      testInState(stateString, requiredState) {
+        // For safety with 'undefined' values, check variety of negation cases:
+        if (stateString == null ||
+            this.stateToDotClassMappings[stateString] == null ||
+            this.stateToDotClassMappings[stateString] != requiredState) {
+          return false;
```

Comment:
> In JS it is safer to use === and !==

Sure, I remember reading about that. But in this case I have just checked & ``null === undefined`` evaluates to ``false``, so I think I might need to change ``null`` -> ``undefined``. I'll take a closer look. I want to make sure all error cases are covered. I guess I am adjusting from my native language, Python, as there isn't a ``KeyError`` equivalent raised in JS for an invalid dict key.

> I think this function can be rewritten

Absolutely, good point (confused as to how I missed that, as it is pure Boolean logic, but I didn't really do a self-review for consolidation etc. at this stage, & think I was just eager to get everything working).

Reply:
I would probably write it like you did too. My IDE warned me about the === and also suggest the rewrite 😁

Reply:
Good old emacs doesn't have that sort of feature. Maybe it is time for me to transition to using Atom :thinking:

Reply:
Actually, for your suggestion on strict (in)equality being safer than abstract (in)equality for comparison, I have just amended the final ``!=``, because (I forgot I already checked this!) in the first pair of cases, comparing in the abstract to ``null`` catches both the ``null`` & ``undefined``, whereas comparing strictly will [only catch one](https://www.w3schools.com/js/tryit.asp?filename=tryjs_datatypes_undefined_3).